os/board/common/crashdump.c

```diff
@@ -204,13 +295,19 @@ void board_crashdump(uint32_t cur_sp, void *tcb, uint8_t *filename, int lineno)
 	up_watchdog_disable();
 #endif
 
+#if defined(CONFIG_BOARD_SMARTFS_DUMP)
+	ret = smartfs_dump();
+	if (ret != OK) {
+		fdbg("SmartFS dump failed, ret = %d\n", ret);
+	}
+#endif
```

Comment:
When this config is enabled, ramdump will not be executed. You should implement yours not to affect others.